python/semantic_kernel/agents/open_ai/responses_agent_thread_actions.py

```diff
@@ -793,6 +817,19 @@ def _get_tool_calls_from_output(
                 )
         return function_calls
 
+    @classmethod
+    def _get_reasoning_items_from_output(
```

Comment:
Response reasoning output consists of interim messages from the model, before the answer is delivered. We'll want to make sure these items are delivered to the caller via the `on_intermediate_message` callback. Our goal is to keep the `invoke` and `invoke_stream` methods delivering "final responses" with interim messages handled via the callback.